tdms/src/fields/base.cpp

```diff
@@ -57,7 +57,88 @@ complex<double> Field::phasor_norm(double f, int n, double omega, double dt, int
          * 1./((double) Nt);
 }
 
+void Field::interpolate_over_range(mxArray **x_out, mxArray **y_out, mxArray **z_out, int i_lower,
+                                   int i_upper, int j_lower, int j_upper, int k_lower, int k_upper,
+                                   Dimension mode) {
+  // prepare output dimensions
+  const int ndims = 3;
+  int outdims[ndims] = {i_upper - i_lower + 1, j_upper - j_lower + 1, 1};
+  if (mode == THREE) {
+    outdims[2] = k_upper - k_lower + 1;
+    if (outdims[1] < 1) {
+      // full simulation (all components being computed) but in 2D - allow one cell in the y-direction to avoid NULL pointers
+      outdims[1] = 1;
+    }
+  }
```

Comment:
This is partially a MATLAB relic and partially because `XYZTensor3D`s members `x`,`y`, and `z` are not `Tensor3D` instances. For some reason, we don't store the array size in the `XYZTensor3D` (although we do in the `Tensor3D` class).

I advocate `x,y,z` to be converted to `Tensor3D` instances in the next refactor too, and `XYZTensor3D` to have a method `fetch_dimensions` or something. `x,y,z` are all designed to be of the same dimensions, so we can just fetch the dimensions of EG `x` when calling this method.